discussment-learning/src/main/java/org/danekja/discussment/learning/service/ArticleService.java

```diff
@@ -0,0 +1,48 @@
+package org.danekja.discussment.learning.service;
+
+import org.danekja.discussment.core.service.DiscussionService;
+import org.danekja.discussment.learning.domain.Article;
+
+import java.util.List;
+
+/**
+ * The interface contains the service methods for working with the articles.
+ *
+ * Date: 30.1.18
+ *
+ * @author Jiri Kryda
+ */
+public interface ArticleService {
+
+
+    /**
+     * Creates a new article and its discussion.
+     *
+     * @param entity new article
+     * @param discussionService instance of the discussion service
+     * @return new article
+     */
+    Article createArticle(Article entity, DiscussionService discussionService);
```

Comment:
DiscussionService to be passed as constructor parameter of the ArticleService implementation, not as method parameter.